### PR TITLE
Add signing server support

### PR DESCRIPTION
**Pull request by enrico-kaack-comp**
**What this PR does / why we need it**:
adds support for signing server sign and verify
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
[Experimental] Add sign signing-server: to sign a component descriptor with a signing server
[Experimental] Add verify x509: to verify a component descriptor with a x509 certificate chain
```

### CODE REVIEW

File: pkg/signatures/sign-server.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	req.SetBasicAuth(signer.Username, signer.Password)
+
+	client := http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed sending request to signing service: %w", err)
+	}
+	defer res.Body.Close()
+	if res.StatusCode != http.StatusOK {
+		if err != nil {
```
Comment (enrico-kaack-comp) [inline pkg/signatures/sign-server.go:57]: fix: if unnecessary

File: pkg/commands/componentarchive/signature/sign/signing_server.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	GenericSignOptions
+}
+
+func NewSigningServerSignCommand(ctx context.Context) *cobra.Command {
+	opts := &SigningServerSignOptions{}
+	cmd := &cobra.Command{
+		Use:   "signing-server BASE_URL COMPONENT_NAME VERSION",
+		Args:  cobra.ExactArgs(3),
+		Short: "fetch the component descriptor from an oci registry and sign it with a signature provided from a signing server",
```
Comment (jschicktanz) [inline pkg/commands/componentarchive/signature/sign/signing_server.go:34]: mention re-upload of cd in docstring

File: pkg/commands/componentarchive/signature/verify/x509_certificate.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+	GenericVerifyOptions
+}
+
+func NewX509CertificateVerifyCommand(ctx context.Context) *cobra.Command {
+	opts := &X509CertificateVerifyOptions{}
+	cmd := &cobra.Command{
+		Use:   "x509 BASE_URL COMPONENT_NAME VERSION",
+		Args:  cobra.ExactArgs(3),
+		Short: "fetch the component descriptor from an oci registry and verify its integrity based on x509 certificates and a RSASSA-PKCS1-V1_5-SIGN signature",
```
Comment (jschicktanz) [inline pkg/commands/componentarchive/signature/verify/x509_certificate.go:38]: "based on x509 cert chain" + use static constant for algorithm

File: pkg/signatures/x509_certificate.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+// CreateAndVerifyX509Certificate creates and verifies a x509 certificate from in-memory raw certificates.
+// The certificates must be in PEM format.
+func CreateAndVerifyX509Certificate(cert, intermediateCAsCerts, rootCACert []byte) (*x509.Certificate, error) {
+	// First, create the set of root certificates. For this example we only
+	// have one. It's also possible to omit this in order to use the
+	// default root set of the current operating system.
+	var roots *x509.CertPool
+	if rootCACert != nil {
+		roots = x509.NewCertPool()
```
Comment (jschicktanz) [inline pkg/signatures/x509_certificate.go:51]: check if this logic is correct

**Review by jschicktanz — APPROVED** [review]

**Pull request merged**